gobblin-utility/src/main/java/gobblin/util/ParallelRunner.java

```diff
@@ -266,6 +268,9 @@ public Void call() throws Exception {
             if (group.isPresent()) {
               HadoopUtils.setGroup(dstFs, dst, group.get());
             }
+            if (commitAction != null) {
+                commitAction.apply();
```

Comment:
Why do you call `apply()` after moving every single file? Shouldn't it only be called once, after all files of a task have been moved?